tests/src/main/java/org/hibernate/beanvalidation/tck/tests/constraints/containerelement/ContainerElementConstraintListTest.java

```diff
@@ -0,0 +1,307 @@
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "listParameter", 0 )
+								.containerElement( "<list element>", true, null, 2, List.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotNull.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "listParameter", 0 )
+								.containerElement( "<list element>", true, null, 2, List.class, 0 )
+						)
+		);
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "a")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "b")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "c")
+	public void constructor_parameter_constraint_provided_on_type_parameter_of_a_list_gets_validated() throws Exception {
+		Constructor<TypeWithList6> constructor = TypeWithList6.class.getDeclaredConstructor( List.class );
+		Object[] values = new Object[] { Arrays.asList( "", "First", null ) };
+
+		Set<ConstraintViolation<TypeWithList6>> constraintViolations = getValidator().forExecutables().validateConstructorParameters(
+				constructor,
+				values
+		);
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithList6.class )
+								.parameter( "listParameter", 0 )
+								.containerElement( "<list element>", true, null, 0, List.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithList6.class )
+								.parameter( "listParameter", 0 )
+								.containerElement( "<list element>", true, null, 2, List.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotNull.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithList6.class )
+								.parameter( "listParameter", 0 )
+								.containerElement( "<list element>", true, null, 2, List.class, 0 )
+						)
+		);
+	}
+
+	private static class TypeWithList1 {
+
+		@SuppressWarnings("unused")
+		private List<@NotNull @NotBlank String> names;
+	}
+
+	private static class TypeWithList2 {
+
+		@Size(min = 1)
+		private List<@NotBlank String> names;
+	}
+
+	private static class TypeWithList3 {
+
+		private List<String> strings;
+
+		@SuppressWarnings("unused")
+		public List<@NotNull @NotBlank String> getStrings() {
+			return strings;
+		}
+	}
+
+	private static class TypeWithList4 {
+
+		private List<String> strings;
+
+		@SuppressWarnings("unused")
+		public List<@NotNull @NotBlank String> returnStrings() {
+			return strings;
+		}
+	}
+
+	private static class TypeWithList5 {
+
+		@SuppressWarnings("unused")
+		public void setValues(@Valid List<@NotNull @NotBlank String> listParameter) {
```

Comment:
Argh, no. The HV tests were doing it wrong. I fixed it everywhere but missed this one. Will fix.